src/lib/eslint/TypescriptEslintStrategy.ts

```diff
@@ -168,75 +120,81 @@ export class TypescriptEslintStrategy implements EslintStrategy {
-		if (!tsconfigPath) {
-			tsconfigPath = await this.checkDirectoryForTsconfig('');
+	/**
+	 * Try to find a tsconfig.json in the current working directory or in the engineOptions.
+	 * Throw an error if exactly one tsconfig.json file can't be found.
+	 */
+	async findTsconfig(engineOptions: Map<string, string>): Promise<string> {
+		const cwd = path.resolve();
+		const tsconfigFromWorkingDirectory = await this.checkWorkingDirectoryForTsconfig();
+		const tsConfigFromOptions = await this.checkEngineOptionsForTsconfig(engineOptions);
+
+		if (!tsconfigFromWorkingDirectory && !tsConfigFromOptions) {
+			// Unable to find in current directory and not specified in the engineOptions
+			throw SfdxError.create('@salesforce/sfdx-scanner', 'TypescriptEslintStrategy', 'MissingTsConfigFromCwd',
+				[TS_CONFIG, cwd, TS_CONFIG]);
+		} else if (tsconfigFromWorkingDirectory && tsConfigFromOptions) {
+			// Found in current directory and specified in the engineOptions
+			throw SfdxError.create('@salesforce/sfdx-scanner', 'TypescriptEslintStrategy', 'MultipleTsConfigs',
+				[TS_CONFIG, tsconfigFromWorkingDirectory, tsConfigFromOptions]);
 		}
 
-		this.logger.trace(`Using ${TS_CONFIG} from ${tsconfigPath}`);
-		return tsconfigPath;
-	}
+		const foundTsConfig = tsconfigFromWorkingDirectory || tsConfigFromOptions;
 
-	private async getTsconfigFromConfig(): Promise<string> {
-		let tsconfigPath = NO_TS_CONFIG;
-		// Check if overriddenConfig is available
-		const overriddenConfig = this.config.getOverriddenConfigPath(ENGINE.ESLINT_TYPESCRIPT);
-		if (!overriddenConfig) {
-			this.logger.trace(`Did not find an overridden path from Config`);
-			return tsconfigPath;
-		}
+		this.logger.trace(`Using ${TS_CONFIG} from ${foundTsConfig}`);
 
-		// Complain if overriddenConfig does not exist.
-		if (!(await this.fileHandler.exists(path.resolve(overriddenConfig)))) {
-			throw SfdxError.create('@salesforce/sfdx-scanner', 'eslintEngine', 'InvalidPath', [overriddenConfig]);
-		}
+		return foundTsConfig;
+	}
 
-		// If overriddenConfig is a directory, look for tsconfig.json file.
-		if (await this.fileHandler.isDir(overriddenConfig)) {
-			this.logger.trace(`Looking for ${TS_CONFIG} inside overridden dir from config ${overriddenConfig}`);
-			tsconfigPath = await this.checkDirectoryForTsconfig(overriddenConfig);
-			// Complain if we can't find a tsconfig.json inside.
-			if (!tsconfigPath) {
-				throw new SfdxError(`Overridden path [${overriddenConfig}] does not contain ${TS_CONFIG}`);
+	/**
+	 * Returns the tsconfig.json file specified in the engineOptions if it is specified and valid, or null if
+	 * engineOptions doesn't specify a tsconfig.json value.
+	 * Throw an error if the file specified in engineOptions is invalid for any reason.
+	 */
+	protected async checkEngineOptionsForTsconfig(engineOptions: Map<string, string>): Promise<string> {
+		const tsConfigFromOptions = engineOptions.get(TYPESCRIPT_ENGINE_OPTIONS.TSCONFIG);
+
+		if (tsConfigFromOptions != null) {
```

Comment:
Have noticed that Typescript returns `undefined` when a key isn't found, unlike Java, which returns `null`. If this already works, I could be wrong.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/get

Reply:
`!= null` covers `null` and `undefined`, which is a bit for most languages, but I thought that was the idiomatic typescript way. 

Would it be better to check for both explicitly with `!==` for null as well as undefined?

Reply:
Ah, thanks, I didn't know that null check covers both `null` and `undefined`.
https://basarat.gitbook.io/typescript/recap/null-undefined#checking-for-either
Let's keep it the way it is then.